smarts/core/smarts.py

```diff
@@ -181,6 +181,13 @@ def step(self, agent_actions):
             self.destroy()
             raise  # re-raise
 
+    def check_if_acting_on_non_existing_agent(self, agent_actions, dones):
+        for agent_id in agent_actions:
+            if agent_id not in dones:
```

Comment:
`if agent_id not in dones` or `if agent_id in dones`? 

Anyway, please see the comments below, I think this checking needs `dones` from the previous step.